### PR TITLE
Enable some zany H(div) elements  (Mardal-Tai-Winther and Arnold-Winther)

### PR DESCRIPTION
**Pull request by rckirby**
This takes UFL (https://github.com/FEniCS/ufl/pull/38 and FIAT), FIAT(https://github.com/FEniCS/fiat/pull/56), and FInAT (https://github.com/FInAT/FInAT/pull/63) live in the form compiler to enable Mardal-Tai-Winther and conforming and nonconforming Arnold-Winther elements.

### CODE REVIEW

File: tsfc/ufl_utils.py [hunk 1]
```diff
@@ -333,6 +333,7 @@ def simplify_abs(expression, complex_mode):
     """Simplify absolute values in a UFL expression.  Its primary
     purpose is to "neutralise" CellOrientation nodes that are
     surrounded by absolute values and thus not at all necessary."""
+
```
Comment (miklos1) [inline tsfc/ufl_utils.py:336]: Unnecessary change.

File: tsfc/driver.py [hunk 2]
```diff
@@ -342,9 +343,15 @@ def compile_expression_dual_evaluation(expression, to_element, coordinates, *,
...
                       ufl_cell=coordinates.ufl_domain().ufl_cell(),
+# <<<<<<< HEAD
+#                       precision=parameters["precision"],
+#                       argument_multiindices=argument_multiindices,
+#                       index_cache={})
+# =======
                       argument_multiindices=argument_multiindices,
                       index_cache={},
                       scalar_type=parameters["scalar_type"])
+# >>>>>>> origin/master
```
Comment (miklos1) [inline tsfc/driver.py:354]: Merge bork.

File: tsfc/driver.py [hunk 3]
```diff
@@ -282,6 +282,7 @@ def compile_expression_dual_evaluation(expression, to_element, coordinates, *,
     :arg parameters: parameters object
     :arg coffee: compile coffee kernel instead of loopy kernel
     """
+
```
Comment (miklos1) [inline tsfc/driver.py:285]: Unnecessary change.

**Review by miklos1 — CHANGES_REQUESTED** [review]

File: tsfc/driver.py (hunk 2 above)
Comment (rckirby) [reply tsfc/driver.py:354]: Good catch, fixed.

File: tsfc/ufl_utils.py (hunk 1 above)
Comment (rckirby) [reply tsfc/ufl_utils.py:336]: Not sure how it crept in.

File: tsfc/driver.py (hunk 3 above)
Comment (rckirby) [reply tsfc/driver.py:285]: Not sure how it happened.  Putting it back.

**Comment by miklos1** [comment]
1. I am somewhat wary of the profusion of methods on `PhysicalGeometry` since interfaces should be minimal, though this is currently an interface with a single implementation (when discounting mock implementations for test cases), so it is not actually a big deal.

2. Please edit the PR title to be more informative.

**Comment by miklos1** [comment]
Pushed a temporary commit for testing with the correct branches...

**Review by miklos1 — APPROVED** [review]

**Comment by miklos1** [comment]
Other than the concerns mentioned [above](https://github.com/firedrakeproject/tsfc/pull/225#issuecomment-676429629), I think this patch is now fine.  Of course, it has to wait until the corresponding branches elsewhere land...

**Review by miklos1 — APPROVED** [review]
This is fine.  I suggest merging with squashing commits, when it eventually comes to merging this.

**Comment by rckirby** [comment]
> This is fine. I suggest merging with squashing commits, when it eventually comes to merging this.

I don't have write access on tsfc, so can you or Lawrence do this on the merge?

**Comment by miklos1** [comment]
> I don't have write access on tsfc, so can you or Lawrence do this on the merge?

Okay, I did it, I hope things won't break...

**Comment by rckirby** [comment]
Thanks.  Fingers crossed..

**Comment by miklos1** [comment]
> I don't have write access on tsfc, so can you or Lawrence do this on the merge?

Though if you can push commits to branches within the TSFC repository, then you do have write access.  It's just that `master` is a protected branch, you cannot merge to master without getting an approve tick from someone else who also has write access.

**Pull request merged**